drivers/SmartThings/matter-thermostat/src/init.lua

```diff
@@ -1089,11 +1119,45 @@ local function set_setpoint(setpoint)
   end
 end
 
-local function setpoint_limit_handler(limit_field)
+local heating_setpoint_limit_handler_factory = function(minOrMax)
+  return function(driver, device, ib, response)
+    if ib.data.value == nil then
+      return
+    end
+    local val = ib.data.value / 100.0
+    device:set_field(minOrMax, val)
+    local min = device:get_field(setpoint_limit_device_field.MIN_HEAT)
+    local max = device:get_field(setpoint_limit_device_field.MAX_HEAT)
+    -- Only emit the capability for RPC version >= 5 (unit conversion for
+    -- heating setpoint range capability is only supported for RPC >= 5)
+    if version.rpc >= 5 and min ~= nil and max ~= nil then
+      if min < max then
+          device:emit_event_for_endpoint(ib.endpoint_id, capabilities.thermostatHeatingSetpoint.heatingSetpointRange({ value = { minimum = min, maximum = max }, unit = "C" }))
+      else
```

Comment:
Same comment as above

Reply:
Changed as suggested in the latest commit!